concurrency-limits-core/src/main/java/com/netflix/concurrency/limits/limit/WindowedLimit.java

```diff
@@ -74,6 +80,17 @@ public Builder minRttThreshold(long threshold, TimeUnit units) {
             return this;
         }
 
+        public Builder percentile() {
```

Comment:
Let's decouple the different metric tracking strategies via a factory and default to average.  So, instead of setting a percentile do something like,
```
WindowedLimit.newBuilder()
    .metricFactory(PercentileSampleWindowFactory.of(0.10))
    .build()
```

Then define a MetricFactory like this,
```
interface SampleWindowFactory {
    SampleWindow create();
}
```

This would also replace `SampleWindow#createBlankInstance()`, which feels a bit awkward given that we're creating a sample window to create a sample window.

Reply:
I agree, would be better, `createBlankInstance()` was an odd one for me as well, but I was unable to come up with a better solution at that moment